src/main/scala/org/tessellation/schema/Hom.scala

```diff
@@ -177,28 +174,78 @@ trait Hom[+A, +B] extends Ω {}
 //    )
 //}
 
+/*
+ *
+ */
 //todo Cv algebras here
-abstract class Cell[M[_]: Monad, F[_]: Traverse, A, B, C](
-  data: A, // L1Edge(txs)
-  // input: G[A], A => G[A]
-  algebra: AlgebraM[M, F, B],
-  coalgebra: CoalgebraM[M, F, C]
-) extends Topos[A, B] {
-  protected def hyloM(implicit input: A => C) = scheme.hyloM(algebra, coalgebra).apply(input(data))
+
+class Cell[M[_]: Monad, F[_]: Traverse, A, B, S](val data: A, val hylo: S => M[B], val convert: A => S)
+    extends Topos[A, B] {
+  def run(): M[B] = hylo(convert(data))
 }
 
 object Cell {
-  //  def create(data, algebra, coalgebra)
-  //  implicit val arrowInstance: Arrow[Cell] = new Arrow[Cell] {
-  //    override def lift[A, B](f: A => B): Cell[A, B] = ???
-  //
-  //    override def first[A, B, C](fa: Cell[A, B]): Cell[(A, C), (B, C)] = ???
-  //
-  //    override def compose[A, B, C](f: Cell[B, C], g: Cell[A, B]): Cell[A, C] =
-  //      // A ---> B ---> C
-  //      f.run(g.run(g)).asInstanceOf[Cell[A, C]]
-  ////      Cell(g.transform).transform
-  //  }
+  case class NullTerminal() extends Ω
+
+  def unapply[M[_], F[_], A, B, S](cell: Cell[M, F, A, B, S]): Some[(A, S => M[B])] = Some(cell.data, cell.hylo)
+
+  implicit def toCell[M[_], F[_], A <: Cell[M, F, Ω, Either[CellError, Ω], Ω]](
+    a: A
+  ): Cell[M, F, Ω, Either[CellError, Ω], Ω] = a.asInstanceOf[Cell[M, F, Ω, Either[CellError, Ω], Ω]]
+
+  implicit def cellMonoid[M[_]: Applicative, F[_]: Applicative](
+    implicit M: Monad[M],
```

Comment:
I think that the `Applicative` constraint on `M` is unnecessary since you have already `Monad` on `M`.

Reply:
Fair point. It must have been a leftover :+1: